binance/websocket/binance_socket_manager.py

```diff
@@ -1,83 +1,100 @@
-import json
 import logging
 import threading
-from urllib.parse import urlparse
-from twisted.internet import reactor, ssl
-from twisted.internet.error import ReactorAlreadyRunning
-from autobahn.twisted.websocket import WebSocketClientFactory, connectWS
-from binance.websocket.binance_client_protocol import BinanceClientProtocol
-from binance.websocket.binance_client_factory import BinanceClientFactory
+from websocket import (
+    ABNF,
+    create_connection,
+    WebSocketException,
+    WebSocketConnectionClosedException,
+)
 
 
 class BinanceSocketManager(threading.Thread):
-    def __init__(self, stream_url):
+    def __init__(
+        self,
+        stream_url,
+        on_message=None,
+        on_open=None,
+        on_close=None,
+        on_error=None,
+        on_ping=None,
+        on_pong=None,
+        logger=None,
+    ):
         threading.Thread.__init__(self)
-
-        self.factories = {}
-        self._connected_event = threading.Event()
+        if not logger:
+            logger = logging.getLogger(__name__)
+        self.logger = logger
         self.stream_url = stream_url
-        self._conns = {}
-        self._user_callback = None
-        self._logger = logging.getLogger(__name__)
-
-    def _start_socket(
-        self, stream_name, payload, callback, is_combined=False, is_live=True
-    ):
-        if stream_name in self._conns:
-            return False
+        self.on_message = on_message
+        self.on_open = on_open
+        self.on_close = on_close
+        self.on_ping = on_ping
+        self.on_pong = on_pong
+        self.on_error = on_error
+        self.create_ws_connection()
 
-        if is_combined:
-            factory_url = self.stream_url + "/stream"
-        else:
-            factory_url = self.stream_url + "/ws"
-
-        if not is_live:
-            payload_obj = json.loads(payload.decode("utf8"))
-
-            if is_combined:
-                factory_url = factory_url + "?streams=" + payload_obj["params"]
-            else:
-                factory_url = factory_url + "/" + payload_obj["params"]
-            payload = None
+    def create_ws_connection(self):
+        self.logger.debug("Start connect to WebSocket Server: %s", self.stream_url)
```

Comment:
Start connect to WebSocket Server
->
Creating connection with WebSocket Server